jax/_src/lax/control_flow/loops.py

```diff
@@ -2020,6 +2021,107 @@ def map(f, xs):
   _, ys = scan(g, (), xs)
   return ys
 
+
+def _lax_map(f, *xs):
+  """Like lax.map, but supports multiple arguments like the built-in map."""
+  g = lambda _, x: ((), f(*x))
+  _, ys = jax.lax.scan(g, (), xs)
+  return ys
+
+def _variable_depth_tree_map(f, tree, *rest):
+  def _tree_map_left(x, *rest):
+    f_wrapper = lambda *rest: f(x, *rest)
+    return tree_map(f_wrapper, *rest)
+
+  return tree_map(_tree_map_left, tree, *rest)
+
+
+def batched_vmap(
+  f: F,
+  in_axes: int | None | Sequence[Any] = 0,
+  out_axes: Any = 0,
+  *,
+  batch_size: int,
+) -> F:
+  """Equivalent to jax.vmap but batches the computation to save memory.
+
+  ``batched_vmap`` splits the ``in_axes`` dimension of the input into batches of
+  size ``batch_size`` and applies the function ``vmap(f)`` to each batch inside
+  a ``scan``.
+
+  Example::
+
+    >>> def f(x):
+    ...   assert x.shape == () # x is a scalar
+    ...   return x ** 2
+    >>> x = np.arange(8)
+    >>> y = jax.lax.batched_vmap(f, batch_size=2)(x)
+    >>> y
+    Array([ 0,  1,  4,  9, 16, 25, 36, 49], dtype=int32)
+
+  Wrapping f with jax.remat() is recommended if you want to use batched_vmap
+  inside gradient calculations. Otherwise, you won't realize any memory savings
+  because JAX will save all the forward activations.
+
+  If your code looks like value_and_grad(mean(vmap(...))) then you probably want
+  to use microbatching instead, which is more efficient because you don't need
+  to use remat.
+
+  Args:
+    f: A Python callable
+    in_axes: An integer, a pytree of integers, or None. Specifies the axes
+      that should be batched. If None, all axes are batched.
```

Comment:
I think `None` already means something else, i.e. don't batch anything.

Can we just delete this last sentence?

```suggestion
      that should be batched.
```